src/stir/mod.rs

```diff
@@ -0,0 +1,74 @@
+pub mod config;
+pub mod ldt;
+pub mod proof;
+pub mod prover;
+pub mod prover_state;
+pub mod verifier;
+pub mod verifier_state;
+
+#[cfg(test)]
+mod tests {
+    use ark_crypto_primitives::sponge::poseidon::PoseidonSponge;
+    use ark_poly::univariate::DensePolynomial;
+    use ark_poly::DenseUVPolynomial;
+    use ark_std::test_rng;
+
+    use crate::{
+        crypto::{fields::Field256, fs, merkle_tree},
+        domain::Domain,
+        ldt::{LowDegreeTest, Prover, Verifier},
+        stir::{config::STIRConfig, ldt::STIR},
+        witness::{
+            single::{SingleWitness, SingleWitnessArgument},
+            Witness,
+        },
+    };
+
+    type TestField = Field256;
+    type TestMerkleConfig = merkle_tree::poseidon::MerkleTreeParams<TestField>;
+    type TestSpongeConfig = PoseidonSponge<TestField>;
+    type TestWitness = SingleWitness<TestField, TestMerkleConfig, TestSpongeConfig>;
+
+    #[test]
+    fn test_stir_ldt() {
+        // config
+        let mut rng = test_rng();
+        let (merkle_leaf_hash_param, merkle_two_to_one_param) =
+            merkle_tree::poseidon::default_config::<Field256>(&mut rng, 2);
+        let config: STIRConfig<TestMerkleConfig, TestSpongeConfig> = STIRConfig {
```

Comment:
@wizardofmenlo you could help me out with some nice params sets if you have time.